Client/Frontend/Browser/TabManager.swift

```diff
@@ -453,14 +453,39 @@ class TabManager: NSObject {
                     for delegate in self.delegates {
                         delegate.get()?.tabManagerDidAddTabs(self)
                     }
+                } else {
+                    self.showFocusPromoToast()
                 }
                 self.eraseUndoCache()
             })
         }
 
         delegates.forEach { $0.get()?.tabManagerDidRemoveAllTabs(self, toast: toast) }
     }
-    
+
+    func showFocusPromoToast() {
+        // add check for event stuff
+        guard let shouldShowFocusUI = LeanplumIntegration.sharedInstance.getBoolVariableFromServer(key: "shouldShowFocusUI"), let focus = NSURL(string: "firefox-focus://"), let klar = NSURL(string: "firefox-klar://"), !UIApplication.shared.canOpenURL(focus as URL) && !UIApplication.shared.canOpenURL(klar as URL) && shouldShowFocusUI else {
```

Comment:
Yea, you are right its not a huge deal.  But I usually try to keep logic isolated. The file TabManager is related to managing tabs. So logic concerning LeanPlum feels like it belongs more in LeanPlumIntegration than in TabManger. 
Also think of the following case. There might be other places where you want to show a focus alert/toast. You’ll also need to check if focus is installed there as well. Instead of having to write this line again you can just do 
`LeanPlumIntegration.sharedInstance.shouldShowFocusInfo()`